install/helm/agones/templates/crds/gameserverallocationpolicy.yaml

```diff
@@ -14,12 +14,10 @@
 
 {{- if .Values.agones.crds.install }}
 
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```

Comment:
@pooneh-m would love for you to give an extra set of eyes to this, just to triple check things :smile: 

I reordered the schema to match the Go structure to make things easier to debug, and also removed some of the extraneous pieces in there.

I expect it all to be fine, and the e2e tests all pass, but figured it was worth an extra check :+1:

Reply:
We don't have any skew tests, but it would be worth doing a quick smoke check installing an old version of agones, running a fleet, and then doing a helm upgrade to see what happens (I think all of our e2e tests assume fresh install).

Reply:
Since our e2e tests do `helm update` - it _should_ be fine, but I can do a quick smoke check, it's a good idea to be sure.

Reply:
Just ran the test - installed Agones 1.10.0 via helm, and then did a `helm update` over the top with this PR's version, and all installed without issue.

Reply:
By default our helm install will remove all previous GameServers on update -- it's quite possible that if you still have a previous GameServer installed, it may not be possible to update via helm without deleting GameServers first. Are we concerned with this scenario? (It's not one we advocate people do in our [update documentation](https://agones.dev/site/docs/installation/upgrading/#upgrading-agones-single-cluster) )

Reply:
I'm not concerned about that scenario, since we tell folks it isn't a supported upgrade path.

Reply:
> I'm not concerned about that scenario, since we tell folks it isn't a supported upgrade path.

Agreed - just wanted to check. :+1:

Reply:
Just checking, is there anything blocking then for this PR, or are we waiting for @pooneh-m to do an extra check?

Reply:
Thanks @markmandel for the change. LGTM.